sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/StringUtilsSuite.scala

```diff
@@ -23,13 +23,27 @@ import org.apache.spark.sql.catalyst.util.StringUtils._
 class StringUtilsSuite extends SparkFunSuite {
 
   test("escapeLikeRegex") {
-    assert(escapeLikeRegex("abdef") === "(?s)\\Qa\\E\\Qb\\E\\Qd\\E\\Qe\\E\\Qf\\E")
-    assert(escapeLikeRegex("a\\__b") === "(?s)\\Qa\\E\\Q_\\E.\\Qb\\E")
-    assert(escapeLikeRegex("a_%b") === "(?s)\\Qa\\E..*\\Qb\\E")
-    assert(escapeLikeRegex("a%\\%b") === "(?s)\\Qa\\E.*\\Q%\\E\\Qb\\E")
-    assert(escapeLikeRegex("a%") === "(?s)\\Qa\\E.*")
-    assert(escapeLikeRegex("**") === "(?s)\\Q*\\E\\Q*\\E")
-    assert(escapeLikeRegex("a_b") === "(?s)\\Qa\\E.\\Qb\\E")
+    assert(escapeLikeRegex("abdef", "\\") === "(?s)\\Qa\\E\\Qb\\E\\Qd\\E\\Qe\\E\\Qf\\E")
+    assert(escapeLikeRegex("abdef", "/") === "(?s)\\Qa\\E\\Qb\\E\\Qd\\E\\Qe\\E\\Qf\\E")
+    assert(escapeLikeRegex("abdef", "\"") === "(?s)\\Qa\\E\\Qb\\E\\Qd\\E\\Qe\\E\\Qf\\E")
+    assert(escapeLikeRegex("a\\__b", "\\") === "(?s)\\Qa\\E\\Q_\\E.\\Qb\\E")
+    assert(escapeLikeRegex("a/__b", "/") === "(?s)\\Qa\\E\\Q_\\E.\\Qb\\E")
+    assert(escapeLikeRegex("a\"__b", "\"") === "(?s)\\Qa\\E\\Q_\\E.\\Qb\\E")
```

Comment:
nit: To make it easy to see, how about this format?
```
    val expectedEscapedStr = "(?s)\\Qa\\E\\Q_\\E.\\Qb\\E"
    assert(escapeLikeRegex("a\\__b", "\\") === expectedEscapedStr)
    assert(escapeLikeRegex("a/__b", "/") === expectedEscapedStr)
    assert(escapeLikeRegex("a\"__b", "\"") === expectedEscapedStr)
```

Reply:
OK. good suggestion.